net/dnscache/dnscache.go

```diff
@@ -202,6 +202,20 @@ func (r *Resolver) LookupIP(ctx context.Context, host string) (ip, v6 netip.Addr
 		r.dlogf("returning %d static results", len(allIPs))
 		return
 	}
+
+	// Hard-code this to avoid extra work, DNS fallbacks, etc.
+	if host == "localhost" {
+		r.dlogf("host is localhost")
+
+		// TODO: @raggi mentioned that some distributions don't use
```

Comment:
on these distros this would be a breaking change where programs have listeners bound by looking up `localhost` before tailscaled starts. I don't think we should TODO this problem.